### PR TITLE
Anna/volunteer dashboard

### PR DESCRIPTION
**Pull request by alin1214**
# Pull Request

## Description

Volunteer Dashboard displaying shifts the user is signed up for (Shift Tasks), and available shifts which leads to Volunteer sign-up page. 

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Other (please describe):

### Code Review Preparation
- [x] I have performed a self-review and reviewed the changed files
- [x] My code follows the project's style guidelines and passes linting
- [x] I have added comments in hard-to-understand areas and updated documentation

### Testing (Ignore for the first PR)
- [ ] I have added relevant tests and they pass locally
- [ ] I have tested the changes manually, including edge cases

### Code Quality & Security
- [ ] I have formatted my code and removed debugging/unused code
- [x] No TypeScript errors or linting issues
- [ ] No sensitive information committed (API keys, passwords, etc.)
- [ ] User inputs are validated and sanitized

### Dependencies & Deployment
- [ ] Dependencies are documented and package-lock.json is updated if needed
- [ ] Changes work with Firebase emulator and environment variables are documentedRetry

## Screenshots (If it is a front end feature screenshot is required)

Screenshot 2026-04-07 at 10 29 55 PM
Screenshot 2026-04-07 at 10 30 13 PM

## Summary by CodeRabbit

* **New Features**
  * Volunteer home page now shows a personalized greeting using your name.
  * Added an "Available Shifts" summary listing up to two upcoming open shifts with dates, volunteer counts vs capacity, status indicators, and sign-up actions.
  * Added a "Shift Tasks" summary showing up to two shifts you’re assigned to with date, status (Open/Upcoming), and quick access to task details.
* **Style**
  * Page layout/background updated for cleaner spacing and contrast.
* **New Icon**
  * Added a right-chevron icon for inline use.

### CODE REVIEW

File: components/homepage/VolunteerHomePage.tsx [hunk 1]
```diff
@@ -1,19 +1,39 @@
...
 import { ProtectedRoute } from "@/components/general/ProtectedRoute";
 import SideNavbar from "@/components/general/SideNav";
+import ShiftTaskSummary from "./ShiftTaskSummary";
+import AvailableShiftsSummary from "./AvailableShiftsSummary";
+import { useAuth } from "@/contexts/AuthContext";
 
 export default function VolunteerHomePage() {
+    // display the name of the user 
+    const { state } = useAuth();
+    const user = state.currentUser;
```
Comment (constrictingsnake) [inline components/homepage/VolunteerHomePage.tsx]: if you get state.userData opposed to state.currentUser, it has more of the user data which is helpful when getting the name field

File: components/homepage/AvailableShiftsSummary.tsx [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+                            0
+                        ) ?? 0;
+
+                    const totalCapacity =
+                        tb.volunteerGroups?.reduce(
+                            (sum, g) => sum + (g.maxNum ?? 0),
+                            0
+                        ) ?? 0;
+
+                    const displayType =
```
Comment (constrictingsnake) [inline components/homepage/AvailableShiftsSummary.tsx]: display type is an unneded value, its okay to just put tb.type in line 100

File: components/homepage/AvailableShiftsSummary.tsx [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+                                    />
+                                    <div className="text-sm">
+                                        {start.toLocaleTimeString([], {
+                                            hour: "numeric",
+                                            minute: "2-digit",
+                                        })}
+                                    </div>
+                                </div>
+
+                                <Link
```
Comment (constrictingsnake) [inline components/homepage/AvailableShiftsSummary.tsx]: the button should match with the other button on the page (in this case it should match the sizing and stuff of the shift task summary component button)

File: components/homepage/AvailableShiftsSummary.tsx [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+                                        volunteers
+                                    </div>
+                                </div>
+                            </div>
+
+                            {/* right */}
+                            <div className="flex flex-col items-end gap-2 ml-auto">
+                                <div className="flex items-center gap-2">
+                                    <div
+                                        className={`w-3 h-3 rounded-full ${
```
Comment (constrictingsnake) [inline components/homepage/AvailableShiftsSummary.tsx]: make sure that the color circle is consistently left aligned across all shifts

File: components/homepage/ShiftTaskSummary.tsx [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+                                        volunteers
+                                    </div>
+                                </div>
+                            </div>
+
+                            {/* right */}
+                            <div className="flex flex-col items-end gap-2 ml-auto">
+                                <div className="flex items-center gap-2">
+                                    <div
+                                        className={`w-3 h-3 rounded-full ${
```
Comment (constrictingsnake) [inline components/homepage/ShiftTaskSummary.tsx]: same as comment on available shifts summary, make sure the circle is consistently left aligned

File: components/homepage/AvailableShiftsSummary.tsx [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+
+            const startTime = tb.startTime.toDate();
+            if (!(tb.published === true && startTime > now)) return false;
+
+            // check if signedup
+            const isSignedUp = tb.volunteerGroups?.some((group) =>
+                group.volunterIDs?.includes(userId)
+            );
+
+            const totalVolunteers =
```
Comment (constrictingsnake) [inline components/homepage/AvailableShiftsSummary.tsx]: the volunteer groups and their capacities should be kept separate; refer to the figma for design

File: components/homepage/VolunteerHomePage.tsx [hunk 7]
```diff
@@ -1,19 +1,39 @@
...
+        "Volunteer";
+
     return (
         <ProtectedRoute allow={["Volunteer"]}>
             <div className="h-full w-full flex flex-col font-family-roboto">
                 <div className="flex flex-1">
                     <SideNavbar />
-                    <div className="flex-1 bg-[#F7F7F7] pt-8 pb-4 px-6 flex flex-col">
+                    <div className="flex-1 bg-[#F7F7F7] pt-8 pb-4 px-6 flex flex-col gap-6">
                         <span className="text-2xl text-primary font-extrabold block">
```
Comment (constrictingsnake) [inline components/homepage/VolunteerHomePage.tsx]: title should align w figma more; go to the figma and press the text to see the typography of the element

File: components/homepage/ShiftTaskSummary.tsx [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+        .sort(
+            (a, b) =>
+                a.startTime.toDate().getTime() - b.startTime.toDate().getTime()
+        );
+
+    return (
+        <div>
+            <Link href="/volunteer-tasks">
+                <div className="flex items-center justify-between mb-4 cursor-pointer">
+                    <h2 className="text-lg text-gray-800">Shift Tasks {">"}</h2>
```
Comment (constrictingsnake) [inline components/homepage/ShiftTaskSummary.tsx]: the > should be an icon; also the text should align with the figma

File: components/homepage/ShiftTaskSummary.tsx [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+                </div>
+            </Link>
+
+            <div className="border bg-white overflow-hidden">
+                {myShifts.map((tb, index) => {
```
Comment (constrictingsnake) [inline components/homepage/ShiftTaskSummary.tsx]: in general for the styling of this component, make sure the font size and weights align with the figma more. the circle for the number is also much bigger than the figma dictates. there should be more space between the date and the type of shift

File: components/homepage/ShiftTaskSummary.tsx [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+                    const start = tb.startTime.toDate();
+                    const isToday = start.toDateString() === now.toDateString();
+
+                    const totalVolunteers =
```
Comment (constrictingsnake) [inline components/homepage/ShiftTaskSummary.tsx]: volunteer groups shouldn't be consolidated, they should be separated into the groups they're in

File: components/homepage/ShiftTaskSummary.tsx [hunk 11]
```diff
@@ -0,0 +1,127 @@
...
+import { useAuth } from "@/contexts/AuthContext";
+
+export default function ShiftTaskSummary() {
+    const { allTB: timeblocks = [] } = useTimeBlocks();
+    const { state } = useAuth();
+    const userId = state.currentUser?.uid;
+
+    const now = new Date();
+
+    const myShifts = timeblocks
```
Comment (constrictingsnake) [inline components/homepage/ShiftTaskSummary.tsx:18]: instead of just having all the tasks show up, we're deciding to limit it to 2 so that the user doesn't have to scroll on most mobile devices

File: components/homepage/AvailableShiftsSummary.tsx [hunk 12]
```diff
@@ -0,0 +1,140 @@
...
+import { useAuth } from "@/contexts/AuthContext";
+
+export default function AvailableShiftsSummary() {
+    const { allTB: timeblocks = [] } = useTimeBlocks();
+    const { state } = useAuth();
+    const userId = state.currentUser?.uid;
+
+    const now = new Date();
+
+    const availableShifts = timeblocks
```
Comment (constrictingsnake) [inline components/homepage/AvailableShiftsSummary.tsx:18]: instead of just having all the tasks show up, we're deciding to limit it to 2 so that the user doesn't have to scroll on most mobile devices

File: components/homepage/AvailableShiftsSummary.tsx [hunk 13]
```diff
@@ -0,0 +1,140 @@
...
+        .sort(
+            (a, b) =>
+                a.startTime.toDate().getTime() - b.startTime.toDate().getTime()
+        );
+
+    return (
+        <div>
+            <Link href="/volunteer-signup">
+                <div className="flex items-center justify-between mb-4 cursor-pointer">
+                    <h2 className="text-lg text-gray-800">Sign-Up {">"}</h2>
```
Comment (constrictingsnake) [inline components/homepage/AvailableShiftsSummary.tsx]: the > should be an icon; also the text should align with the figma

File: components/homepage/AvailableShiftsSummary.tsx [hunk 14]
```diff
@@ -0,0 +1,140 @@
...
+                            (sum, g) => sum + (g.maxNum ?? 0),
+                            0
+                        ) ?? 0;
+
+                    const displayType =
+                        tb.type === "Pickup/Delivery"
+                            ? "Pickups / Deliveries"
+                            : "Warehouse";
+
+                    return (
```
Comment (constrictingsnake) [inline components/homepage/AvailableShiftsSummary.tsx]: in general, with regards to styling it's the same comments as the one for shift tasks

File: components/homepage/VolunteerHomePage.tsx [hunk 15]
```diff
@@ -1,19 +1,39 @@
...
+        user?.email?.split("@")[0] ||
+        "Volunteer";
+
     return (
         <ProtectedRoute allow={["Volunteer"]}>
             <div className="h-full w-full flex flex-col font-family-roboto">
                 <div className="flex flex-1">
                     <SideNavbar />
-                    <div className="flex-1 bg-[#F7F7F7] pt-8 pb-4 px-6 flex flex-col">
+                    <div className="flex-1 bg-[#F7F7F7] pt-8 pb-4 px-6 flex flex-col gap-6">
```
Comment (constrictingsnake) [inline components/homepage/VolunteerHomePage.tsx]: the background is supposed to be white

File: components/homepage/ShiftTaskSummary.tsx [hunk 16]
```diff
@@ -0,0 +1,127 @@
...
+            <Link href="/volunteer-tasks">
+                <div className="flex items-center justify-between mb-4 cursor-pointer">
+                    <h2 className="text-lg text-gray-800">Shift Tasks {">"}</h2>
+                </div>
+            </Link>
+
+            <div className="border bg-white overflow-hidden">
+                {myShifts.map((tb, index) => {
+                    const start = tb.startTime.toDate();
+                    const isToday = start.toDateString() === now.toDateString();
```
Comment (constrictingsnake) [inline components/homepage/ShiftTaskSummary.tsx]: the open button should only show when the shift starts, ie make sure that the current time is within the start and end times

**Review by constrictingsnake — CHANGES_REQUESTED** [review]

**Review by joelchem — APPROVED** [review]
